packages/tpetra/core/src/Tpetra_BlockCrsMatrix_def.hpp

```diff
@@ -968,6 +972,7 @@ class GetLocalDiagCopy {
          prefix << "The matrix's values need sync on both device and host.");
 #endif // HAVE_TPETRA_DEBUG
       this->modify_host ();
+      // DEEP_COPY REVIEW - NOT TESTED
```

Comment:
The method is setAllToScalar. The majority of use cases in my opinion is to set the matrix zero. Otherwise, setting all block crs matrix with a scalar value does not make sense to me. Then, why do we need to write this complicated routine ? We just call deep copy of scalar value for both host and device and set the dual view is synced. This is probably easier.